### PR TITLE
Fix back button not working in some activities.

### PR DESCRIPTION
**Pull request by yunikkk**
Back button was not working in some activities since we hide it via onOptionsItemSelected returning true

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: app/src/main/java/com/mapbox/maps/testapp/examples/DebugModeActivity.kt [hunk 1]
```diff
@@ -53,6 +54,13 @@ class DebugModeActivity : AppCompatActivity() {
     mapboxMap.subscribe(observable, listOf(MapEvents.RESOURCE_REQUEST))
     // Using the extension method
     mapboxMap.subscribeResourceRequest(extensionObservable)
+    mapboxMap.loadStyle(style("") {})
```
Comment (alexander-kulikovskii) [inline app/src/main/java/com/mapbox/maps/testapp/examples/DebugModeActivity.kt:57]: do we really need such setup?

**Review by alexander-kulikovskii — APPROVED** [review]

**Comment by ank27** [comment]
💎

File: app/src/main/java/com/mapbox/maps/testapp/examples/DebugModeActivity.kt (hunk 1 above)
Comment (yunikkk) [reply app/src/main/java/com/mapbox/maps/testapp/examples/DebugModeActivity.kt:57]: Nooo, that remainder from test setup, will remove

File: app/src/main/java/com/mapbox/maps/testapp/examples/DebugModeActivity.kt [hunk 2]
```diff
@@ -8,6 +8,7 @@ import com.mapbox.maps.*
 import com.mapbox.maps.extension.observable.eventdata.MapLoadingErrorEventData
 import com.mapbox.maps.extension.observable.getResourceEventData
 import com.mapbox.maps.extension.observable.subscribeResourceRequest
+import com.mapbox.maps.extension.style.style
```
Comment (alexander-kulikovskii) [inline app/src/main/java/com/mapbox/maps/testapp/examples/DebugModeActivity.kt:11]: unused import

Comment (yunikkk) [reply app/src/main/java/com/mapbox/maps/testapp/examples/DebugModeActivity.kt]: Not anymore

**Pull request merged**